statefun-kafka-io/src/main/java/org/apache/flink/statefun/sdk/kafka/KafkaIngressSpec.java

```diff
@@ -75,12 +72,41 @@ public KafkaIngressStartupPosition startupPosition() {
     return startupPosition;
   }
 
-  private static void validateStartupPosition(
-      KafkaIngressStartupPosition startupPosition, @Nullable String groupIdConfig) {
-    if (startupPosition.isGroupOffsets() && groupIdConfig == null) {
+  private static Properties requireValidProperties(Properties properties) {
```

Comment:
@tzulitai What are your thoughts on having the validation happen in the builder?

Reply:
The only "downside" I had in mind when not implementing this in the builder was that it is slightly more dangerous implementing it there, because the timeliness of when the validation was done will affect the end result (before / after the properties resolution).
Just add one little bit of space to go wrong, although very trivial.
Being done in the `KafkaIngressSpec`, this should not be the case.

Reply:
I don't have a strong stand on this though.
For a better separation of responsibility wise, it could make sense to still move this to the builder yes.

Reply:
I don't have a strong stand on this though.
For a better separation of responsibility wise, it could make sense to still move this to the builder yes.

Reply:
I'd like to keep this as is for now, but am not against changing it.
Please let me know if you'd like this changed after merging.